### PR TITLE
RA-1928: LoginPageController to handle URLs with one "/" only.

### PR DESCRIPTION
**Pull request by icrc-jfigueiredo**
RA-1928 If only contains one "/" character, it will use the entire urlPath for the urlContextPath
Issue: https://issues.openmrs.org/browse/RA-1928

### CODE REVIEW

**Comment by HerbertYiga** [comment]
thanks @icrc-jfigueiredo  you can also add the ticket id here

**Review by mks-d — APPROVED** [review]

File: omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ private boolean isUrlWithinOpenmrs(PageRequest pageRequest, String redirectUrl)
 				try {
 					URL url = new URL(redirectUrl);
 					String urlPath = url.getFile();
-					String urlContextPath = urlPath.substring(0, urlPath.indexOf('/', 1));
+					String urlContextPath = urlPath.substring(1).contains("/") ? urlPath.substring(0, urlPath.indexOf('/', 1)) : urlPath;
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java:165]: It would be great to add a test for this.

Comment (icrc-jfigueiredo) [reply omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java:165]: Hello @dkayiwa , sure, I just added a new test for this. Thank you.

Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java:165]: @icrc-jfigueiredo did you notice that your test passes even without the above changes?

Comment (icrc-jfigueiredo) [reply omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java:165]: I did not notice that, but if we try to use something different from "/openmrs" like "/test" on the URL the test will fail.

Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java:165]: @icrc-jfigueiredo all am trying to say is simply that you are not testing the change. For the test would start by failing and then your changes make it pass.

Comment (icrc-jfigueiredo) [reply omod/src/main/java/org/openmrs/module/referenceapplication/page/controller/LoginPageController.java:165]: @dkayiwa I updated the test to only pass with my modification, now it really tests the new development.

File: omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java [hunk 2]
```diff
@@ -334,6 +334,33 @@ public void get_shouldRedirectUserToHomeIfAuthenticated() throws Exception {
...
+	 *      org.openmrs.module.appframework.service.AppFrameworkService,
+	 *      org.openmrs.api.AdministrationService)
+	 */
+	@Test
+	@Verifies(value = "should redirect user to home when user is authenticated if ? redirectUrl is for openrms, even if only have  '/openrms'", method = "get(PageModel,UiUtils,PageRequest)")
+	public void get_shouldRedirectUserToHomeIfAuthenticatedAndPathForOpenmrsWasGivenOnUrl() throws Exception {
+		when(Context.isAuthenticated()).thenReturn(true);
+		when(Context.getUserContext()).thenReturn(mock(UserContext.class));
+
+		String redirectFullUrl = "http://openmrs.org/openmrs";
```
Comment (dkayiwa) [inline omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java:351]: Are these variables or constants?

File: omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java [hunk 3]
```diff
@@ -334,6 +334,33 @@ public void get_shouldRedirectUserToHomeIfAuthenticated() throws Exception {
...
+	/**
+	 * @see LoginPageController#get(org.openmrs.ui.framework.page.PageModel,
+	 *      org.openmrs.ui.framework.UiUtils, org.openmrs.ui.framework.page.PageRequest, String,
+	 *      org.openmrs.api.LocationService,
+	 *      org.openmrs.module.appframework.service.AppFrameworkService,
+	 *      org.openmrs.api.AdministrationService)
+	 */
+	@Test
+	@Verifies(value = "should redirect user to home when user is authenticated if ? redirectUrl is for openrms, even if only have  '/openrms'", method = "get(PageModel,UiUtils,PageRequest)")
+	public void get_shouldRedirectUserToHomeIfAuthenticatedAndPathForOpenmrsWasGivenOnUrl() throws Exception {
```
Comment (dkayiwa) [inline omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java:347]: Can we also improve the test method name?

File: omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java (hunk 2 above)
Comment (icrc-jfigueiredo) [reply omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java:351]: Constants, I update according to the constants convention.

File: omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java (hunk 3 above)
Comment (icrc-jfigueiredo) [reply omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java:347]: Absolutely, do you have some suggestions? maybe something like:
get_shouldRedirectUserToHomeIfAuthenticatedAndPathUrlAreInOpenmrsScope
get_shouldRedirectUserToHomeIfAuthenticatedAndUrlsForOpenmrs

Comment (dkayiwa) [reply omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java:347]: Pick what you see best.

Comment (icrc-jfigueiredo) [reply omod/src/test/java/org/openmrs/module/referenceapplication/page/controller/LoginPageControllerTest.java]: @dkayiwa i renamed it to `get_shouldRedirectUserToHomeIfAuthenticatedAndURLPathAreInOpenmrsScope`

**Pull request merged**